controllers/user.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/lifeiscontent/webservice/models"
+)
+
+type userController struct {
+	userIDPattern *regexp.Regexp
+}
+
+func (uc userController) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path == "/users" {
+		switch r.Method {
+		case http.MethodGet:
+			uc.getAll(w, r)
+		case http.MethodPost:
+			uc.post(w, r)
+		default:
+			w.WriteHeader(http.StatusNotImplemented)
+		}
+	} else {
+		matches := uc.userIDPattern.FindStringSubmatch(r.URL.Path)
+		if len(matches) == 0 {
+			w.WriteHeader(http.StatusNotFound)
+		}
+		id, err := strconv.Atoi(matches[1])
+		if err != nil {
+			w.WriteHeader(http.StatusNotFound)
+		}
+		switch r.Method {
+		case http.MethodGet:
+			uc.get(id, w)
+		case http.MethodPut:
+			uc.put(id, w, r)
+		case http.MethodDelete:
+			uc.delete(id, w)
+		default:
+			w.WriteHeader(http.StatusNotImplemented)
```

Comment:
@zigara bugs the crap out of me that we have to use tabs lol

Reply:
This is one of the best things about Go, there are no debates on formatting, you run gofmt on your code and everyone else does too. You'll get used to it and miss it in other languages, especially the structure field alignment.

Reply:
yeah, I noticed the struct field alignment, pretty cool stuff for sure.